core/src/crds_gossip_pull.rs

```diff
@@ -371,15 +372,47 @@ impl CrdsGossipPull {
         &self,
         crds: &Crds,
         filters: &[(CrdsValue, CrdsFilter)],
+        now: u64,
     ) -> Vec<Vec<CrdsValue>> {
         let mut ret = vec![vec![]; filters.len()];
+        let msg_timeout = CRDS_GOSSIP_PULL_CRDS_TIMEOUT_MS;
+        let start = filters.len();
+        //skip filters from callers that are too old
+        let recent: Vec<_> = filters
+            .iter()
+            .filter(|(caller, _)| {
+                caller.wallclock() < now.checked_add(msg_timeout).unwrap_or_else(|| 0)
+                    && caller
+                        .wallclock()
+                        .checked_add(msg_timeout)
+                        .unwrap_or_else(|| 0)
+                        >= now
+            })
+            .collect();
+        inc_new_counter_info!(
+            "gossip_filter_crds_values-dropped_requests",
+            start - recent.len()
+        );
+        if recent.is_empty() {
+            return ret;
+        }
+        let mut val = 0;
         for v in crds.table.values() {
-            filters.iter().enumerate().for_each(|(i, (_, filter))| {
-                if !filter.contains(&v.value_hash) {
-                    ret[i].push(v.value.clone());
-                }
-            });
+            filters
+                .iter()
+                .enumerate()
+                .for_each(|(i, (caller, filter))| {
+                    //skip values that are too new
+                    if v.value.wallclock() > caller.wallclock() {
```

Comment:
@carllin @sakridge if the source is constantly updating this value, then this might always fail

Reply:
wdyt of
```
if v.value.wallclock() > caller.wallclock().checked_add(rng:range(0, msg_timeout)) {
}
```

Reply:
hmmm with msg_timeout = 15 seconds, then on average this timeout with be 7 seconds which seems awfully long (seems like it won't filter out very much). It seems like this delay should be based on expected number of hops for propagation of push messages through the network? Something like `log(network_size) * 150ms`.

Reply:
hmmm with msg_timeout = 15 seconds, then on average this timeout with be 7 seconds which seems awfully long (maybe won't filter out very much?). It seems like this delay should be based on expected number of hops for propagation of push messages through the network? Something like `log(network_size) * 150ms`.

Reply:
@carllin Caller is expected to be 3.5 seconds old.  I guess i can set the jitter to random between 0 and msg_timeout/2

Reply:
i want gossip to be able to eventually get all the data through pulls, otherwise we could get hosed

Reply:
ah the caller is the node's ContactInfo which is refreshed in `push_self()` every `CRDS_GOSSIP_PULL_CRDS_TIMEOUT_MS / 2` = 7, so we expect the caller to be around 3.5 seconds old on average.

So essentialy what we're saying is, every time the node updates their ContactInfo at time `X`, only receive messages through pull requests with timestamp < `X + Y`, where Y is the time until the next update of `ContactInfo`, at which point we'll get those newer messages we're missing. Cool!

Is there any reason to use jitter instead of a fixed `CRDS_GOSSIP_PULL_CRDS_TIMEOUT_MS / 2` or `CRDS_GOSSIP_PULL_CRDS_TIMEOUT_MS / 4`?